### DISCUSSION TITLE
Error reading glb file

### DISCUSSION THREAD

**Discussion by freyatech** [donmccurdy/glTF-Transform#366] [Q&A] [answered]
Hi! I'm trying to read glb file with NodeIO read but I keep getting an error:
"Uncaught (in promise) Error: Cannot read properties of undefined (reading 'DT_FLOAT32')".
Any ideas what I might be doing wrong? gltf-transform/cli is able to read and transform the file.

thanks for any info.

**Comment by donmccurdy** [comment, marked as answer]
`DT_FLOAT32` is a Draco constant, it sounds like the file you're trying to load might require Draco decompression? If so, you'd need to use the `draco3dgltf` npm package and register it as shown here:

https://gltf-transform.donmccurdy.com/classes/extensions.dracomeshcompression.html#example

```js
import { NodeIO } from '@gltf-transform/core';
import { KHRONOS_EXTENSIONS } from '@gltf-transform/extensions';
import draco3d from 'draco3dgltf';

// ...

const io = new NodeIO()
    .registerExtensions(KHRONOS_EXTENSIONS)
    .registerDependencies({
        'draco3d.decoder': await draco3d.createDecoderModule(), // Optional.
        'draco3d.encoder': await draco3d.createEncoderModule(), // Optional.
    });
```

I had thought that NodeIO gave a clearer error when this dependency was missing though... will take a look at that.

**Comment by freyatech** [comment]
Hi Don!

Thank you for your reply. I’ve tried to implement it both in node and front end but keep getting the same error:

Uncaught (in promise) RuntimeError: abort(CompileError: WebAssembly.instantiate(): expected magic word 00 61 73 6d, found 3c 21 44 4f @+0). Build with -s ASSERTIONS=1 for more info.

Any ideas?

Thanks
Lucas

**Reply by donmccurdy** [reply]
The `draco3dgltf` package will only work in Node.js I think. For use on the frontend see [this answer](https://stackoverflow.com/a/66979159/1314762). Unfortunately I can't tell much from that error, can you share enough of your code (and build process, if any?) for me to reproduce that?

**Comment by freyatech** [comment]
Hi!

What I'm trying to run it on is electron app with react as UI. As I'm quite new to electron development I'm not sure whether electron's main process is classed as node environment or the whole setup is classed as front end since it's running in Chrome. 
I've still got few ideas that I want to try, but if that clears the issue for you and you've got some suggestions I'd be extremely grateful.

Regards
Lucas

⁣Sent from BlueMail ​

**Reply by donmccurdy** [reply]
Hm, that's a great question. I would think either WebIO or NodeIO could work in Electron, but I'm not sure. If you _just_ include `await draco3d.createDecoderModule()`, does that alone throw an error? The second error (`WebAssembly.instantiate()`) is something about the Draco library failing to load its WASM dependencies. :/
